### DISCUSSION TITLE
Cercube Developer [Discussion]

### DISCUSSION THREAD

**Discussion by arichornlover** [arichornlover/YouTubeRebornPlus#19] [General]
Hello everyone. I've went on reddit to see what happened to Cercube and I found out Google found out about Cercube and I think that's the reason why Cercube is gone. here's the problem about this, none of y'all will not be able to download stuff on cercube so I will give y'all a choice in the reply below.

**Comment by arichornlover** [comment]
POLL:
Should I Disable the **Cercube Download Tab** by Default?
you can reply below since it may be useless in the future.

**Comment by Ketarmin** [comment]
Yea but with a popup or something else to know why you did this 

mh that’s maybe why YouTube start’s crashing so suddenly but it works with the Cer v5.3.11 

Can you share the link of the Reddit post’s pls?

**Comment by arichornlover** [comment]
> Yea but with a popup or something else to know why you did this

Sure. I'll try to add a popup
> mh that’s maybe why YouTube start’s crashing so suddenly but it works with the Cer v5.3.11

is 5.3.12 just broken because I thought it was bug fixes for the new yt ui?

> Can you share the link of the Reddit post’s pls?

https://www.reddit.com/r/sideloaded/comments/zm3ts3/cercube_on_ios_keeps_crashing/
https://www.reddit.com/r/sideloaded/comments/zox6v9/cercube_is_dead_long_live_the_king_uyou/
https://www.reddit.com/r/sideloaded/comments/zpflm1/cercubeyoutube_alts_crashing_easy_solution/

**Comment by Ketarmin** [comment]
> is 5.3.12 just broken because I thought it was bug fixes for the new yt ui?

Hm I don’t think so that it’s broke, I think, if it’s true that Google coursing this crashes then they are scanning for the latest Cer version. 

What make me think that is because when I use the last Cer+ version Qn released which still have had Cer v5.3.11 in it, it works fine but if I injected tweaks with your fork in to the same YT version, which was 17.43.1, the version Qn used in his last release than it starts crashing too ( I hope I explained it somehow understandable xD )

like:

Qn:  Cer+ v5.3.11 on YT v17.43.1 = no crashes 

your latest release starts suddenly crashing on YT v17.49.4 with 5.3.12

my test build with your fork with Cer v5.3.12 on YT v17.43.1 now crashes too but Qn build as mentioned above doesn’t crash.

>is 5.3.12 just broken because I thought it was bug fixes for the new yt ui?

it have just few bug fixes but as I tried to explain above it unfortunately crashes now
 :(


Thx for the links.

**Comment by arichornlover** [comment]
hmm so you want me to downgrade it to cer 5.3.11?

**Comment by arichornlover** [comment]
this won't fix the download error but it'll hopefully fix the crashes right?

**Comment by Ketarmin** [comment]
> hmm so you want me to downgrade it to cer 5.3.11?

Yea, but only if it fixes the crashing issue 

> this won't fix the download error but it'll hopefully fix the crashes right?

I think maybe Majd Alfhaily (Cer dev) had to take down the download servers

**Comment by arichornlover** [comment]
also I came back I failed to implement a popup for CercubePlus so I putted the reason of disabling the download tab under the hideCercubeDownload Option

**Comment by Ketarmin** [comment]
> also I came back I failed to implement a popup for CercubePlus so I putted the reason of disabling the download tab under the hideCercubeDownload Option

Thank you very much, i think that’s even better because it doesn’t distract the user but still explains it nicely

**Comment by arichornlover** [comment]
> Thank you very much, i think that’s even better because it doesn’t distract the user but still explains it nicely

No problem!

**Comment by arichornlover** [comment]
Also I just realized since the Cercube developer is now offline and no longer updating their tweak, I think eventually Cercube won’t support any new YouTube versions in a few more years if Google messes with the YouTube code then pretty much once the app version gets unsupported by google we will might aswell move on from Cercube and eventually replace it with either `YouTube Murderer`, `YouTube Reborn` or `uYou`/`uYouPlus`/`uYouPlusExtra`

**Comment by rowelln12** [comment]
Hey thanks for all the hard work! I was wondering, is there a good work around for not being able to sign in. Google says the app is unsafe so it wont let me sign into my account.

**Comment by arichornlover** [comment]
> Hey thanks for all the hard work! I was wondering, is there a good work around for not being able to sign in. Google says the app is unsafe so it wont let me sign into my account.

https://github.com/qnblackcat/uYouPlus/discussions/447
This should help you

**Comment by rowelln12** [comment]
> > Hey thanks for all the hard work! I was wondering, is there a good work around for not being able to sign in. Google says the app is unsafe so it wont let me sign into my account.
> 
> [qnblackcat/uYouPlus#447](https://github.com/qnblackcat/uYouPlus/discussions/447) This should help you

Worked like a champ, thanks a million!

**Comment by arichornlover** [comment]
No Problem!

**Comment by Ketarmin** [comment]
> Worked like a champ, thanks a million!

Nice that it worked, if it some day doesn’t work then try this 

https://github.com/qnblackcat/uYouPlus/discussions/447#discussioncomment-3672653

this worked for me when the other options doesn’t

**Comment by arichornlover** [comment]
![D147A811-1940-4320-A60F-79B4F581A164](https://user-images.githubusercontent.com/78001398/210287985-64df68e2-5b04-4a9c-9cd1-06cc141388eb.jpeg)
This is what settings in the CercubePlus Section will look like, even though this is uYouPlus, the settings will look like this in the new release either way 👍

**Comment by arichornlover** [comment]
I did this change to push out the newer stuff based off of uYouPlus & YouTube Reborn to make CercubePlus great again!

**Comment by Ketarmin** [comment]
> I did this change to push out the newer stuff based off of uYouPlus & YouTube Reborn to make CercubePlus great again!

Alright

**Comment by Ketarmin** [comment]
Hey @arichorn, do you have any connection to @poomsmart ? 

If you have then you could may ask him if he is able to finde a solution for the Cer v5.3.12 problem, where it crashes?

**Comment by arichornlover** [comment]
Well I’m not connected to PoomSmart but I can still ask him.

**Comment by dayanch96** [comment]
[cercube_fix.zip](https://github.com/arichorn/CercubePlusExtra/files/10609829/cercube_fix.zip)

No need to disable it. Here is an archive with my Cercube and fix which allows you to download videos and also brings back albums and avatars for users from Russia. Just inject this YouTube_Albums_Fix.dylib and Cercube.dylib instead of your current Cercube.dylib and keep original bundle and zCercubeBoard if you injecting it.

**Comment by dayanch96** [comment]
> where it crashes?

it crashes when gets response from cercube api that you're using sideloaded version

**Comment by Ketarmin** [comment]
> [cercube_fix.zip](https://github.com/arichorn/CercubePlusExtra/files/10609829/cercube_fix.zip)
> 
> No need to disable it. Here is an archive with my Cercube and fix which allows you to download videos and also brings back albums and avatars for users from Russia. Just inject this YouTube_Albums_Fix.dylib and Cercube.dylib instead of your current Cercube.dylib and keep original bundle and zCercubeBoard if you injecting it.

Does it even work with the Cercube v5.3.12 in combination with the latest YT ipa ?

**Comment by arichornlover** [comment]
> it crashes when gets response from cercube api that you're using sideloaded version

Downgrading to Cercube v5.3.11 fixes the crash problem so I’m confused.

**Comment by dayanch96** [comment]
> 

it has inside modified cercube, which is theoretically 5.3.12

**Comment by dayanch96** [comment]
unzip and inject

[me.alfhaily.cercube_5.3.13_iphoneos-arm.zip](https://github.com/arichorn/CercubePlusExtra/files/10771467/me.alfhaily.cercube_5.3.13_iphoneos-arm.zip)

**Reply by ssuesskind** [reply]
Thank you for providing this, i injected it successfully into CercubePlusExtra using YouTube 18.11.2.

However, I am getting an `Error 500` when trying to download Videos.

And another question: are the sources publicly available?

Kind regards

**Reply by arichornlover** [reply]
> And another question: are the sources publicly available?

Cercube is not Open-Source sadly

> Thank you for providing this, i injected it successfully into CercubePlusExtra using YouTube 18.11.2.

I’m very impressed about the issue being resolved. Thanks for resolving the last few errors in it 😇 I’ll see if I have the time to update the README.md and try to put you in the credits

**Reply by ssuesskind** [reply]
Thank you, you're welcome. 😁
Looking forward to contribute more things.

**Reply by arichornlover** [reply]
> [me.alfhaily.cercube_5.3.13_iphoneos-arm.zip](https://github.com/arichorn/CercubePlusExtra/files/10771467/me.alfhaily.cercube_5.3.13_iphoneos-arm.zip)

Hey @dayanch96 ! I'm very sorry for disturbing you but can the `YouTubeReborn.dylib` be injected in the DynamicLibraries Folder? [youtube-reborn.tar.gz](https://github.com/arichorn/CercubePlusExtra/files/11338972/youtube-reborn.tar.gz)

I'm adding another tweak to basically give sideloaded users ability to download videos if Cercube Servers don't work for them.
I'm also gonna modify Cercube.Dylib on Windows to fix Dark Mode Issue and Options if possible. just wanting to make CercubePlus have lesser issues and some additional features. I'm really trying my best to make this Tweak better. and Dayanch, you are one of the best contributors to help improve CercubePlus and I really appreciate it!

**Comment by Ketarmin** [comment]
> unzip and inject
> 
> [me.alfhaily.cercube_5.3.13_iphoneos-arm.zip](https://github.com/arichorn/CercubePlusExtra/files/10771467/me.alfhaily.cercube_5.3.13_iphoneos-arm.zip)

Sooo alfhaily is still maintaining Cercube or am I completely wrong?

**Reply by arichornlover** [reply]
hmm, just coming back to this to say that I could do a `5.3.14` mod to fix the 2 Cercube Options "Hide Cast Button" & "Block Ads" but I have no Mac/Xcode so me modifying the deb isn't gonna likely happen. I had this in mind because there were some issues that aren't caused by CercubePlus.

**Reply by Ketarmin** [reply]
„Block ads“ isn’t it working correctly? Because I have no problems with it

**Reply by arichornlover** [reply]
I’ve seen issues with it not working time-to-time when I was testing CercubePlus on the release `18.13.4-5.3.13` which is why I added this to the latest just to solve it https://github.com/arichorn/CercubePlusExtra/blob/4de009839f10291c846b8a8e41fc5ef258bd1898/CercubePlus.xm#L232

**Comment by dayanch96** [comment]
> am I completely wrong?

its my mod for 5.3.12

**Comment by arichornlover** [comment]
Thanks Dayanch you literally just saved CercubePlus ☺️✌️

**Comment by Ketarmin** [comment]
> > am I completely wrong?
> 
> its my mod for 5.3.12

Ohhhh nice, thank you very much

**Comment by arichornlover** [comment]
https://github.com/arichorn/CercubePlusExtra/actions/runs/4208124144/jobs/7303771854
well, sadly I’m still having the same building errors, This error indicates the deb file just downloads and gets corrupted when building. I was gonna release 18.06.6-5.3.11 or 18.06.6-5.3.13 today but I’m just trying to not let y’all down on how bad of a job of I’m doing

**Comment by dayanch96** [comment]
> https://github.com/arichorn/CercubePlusExtra/actions/runs/4208124144/jobs/7303771854 
oops, looks wrong compression. try this one

[me.alfhaily.cercube_5.3.13_iphoneos-arm.zip](https://github.com/arichorn/CercubePlusExtra/files/10772877/me.alfhaily.cercube_5.3.13_iphoneos-arm.zip)

**Comment by dayanch96** [comment]
> I don’t get why it’s corrupting the deb

Idk, maybe because you're trying to download 5.3.13 from Majd's repo? :D

**Comment by dayanch96** [comment]
just place deb file into Tweaks/Cercube folder and remove curl line in actions file.
also note that the action used for the main branch and your edits in main-testing branch

**Comment by arichornlover** [comment]
Yeah. I needed to use the main-testing branch to not flood the main branch with the failed attempts of fixing and building the deb, I still have no luck but I don’t want to let CercubePlusUsers down. And I removed the curl line.

**Comment by arichornlover** [comment]
Hey Dayanch! It turns out I didn’t modify buildapp.yml, that file was the one that had the Cercube Deb and I did change it back to 5.3.11 but the reason for that is because it didn’t recognize your modded Cercube as a Debian file. But it works when I build 5.3.11.

**Reply by dayanch96** [reply]
Something wrong on your side

**Reply by arichornlover** [reply]
Well here’s the error of me putting the modified deb file into CercubePlus https://github.com/arichorn/CercubePlusExtra/actions/runs/4257948991/jobs/7408638015
It’s fine though, I do really appreciate you helping though.

**Reply by dayanch96** [reply]
Hmm, try to unpack it manually to be sure

**Reply by arichornlover** [reply]
![D68781BB-3B18-4DC2-B03A-D17030CD99A5](https://user-images.githubusercontent.com/78001398/221451061-17cd2c7d-0866-43cf-aff1-a263bfa4bb80.jpeg)
![D6876F20-BE96-42FF-AB4D-83A409375E6A](https://user-images.githubusercontent.com/78001398/221451062-9204d141-80ec-4b03-bda0-e680c1c1e58c.jpeg)
I extracted them and this is what I’ve see. Hope it helps

**Reply by dayanch96** [reply]
how did you extracted it?

**Reply by arichornlover** [reply]
I used a site called https://createzip.com/extract/deb
and it really helped out
I was gonna do it manually but I wasn’t on the computer so had to use the site as an alternative.

**Reply by dayanch96** [reply]
just use dpkg -x /your.deb /output/folder (or even _ar -x /your.deb_ if you dont want to install dpkg)

if your building script requires tar.gz and tar.lzma - then just repack it or use already extracted files.

**Comment by Ketarmin** [comment]
Please stay, I love your hard work 😘😘

**Comment by arichornlover** [comment]
## This Issue has been migrated into a Discussion
Doing this was very necessary because this wasn’t technically an issue so decided to make it a Discussion.

**Reply by arichornlover** [reply]
⚠️to anyone reading this message, CercubePlusExtra is getting de-forked which means this whole entire Repository will be converted into a non-forked repository which also removes UprisingPulse’s Repo from CercubePlusExtra. I’m only saying this because this type of change might take effect on the releases & your forks of CercubePlusExtra.⚠️

**Reply by Ketarmin** [reply]
Why you doing this?

**Reply by arichornlover** [reply]
Because I’m separating CercubePlusExtra from all the other CercubePlus forks because it would be confusing for others who come across this repo but hopefully my decision makes it a little safer for this repo because I can now make changes and improvements without accidentally pressing the “sync fork” button where I might mistakenly delete all of the commits.

and the change already happened thanks to the GitHub team. 👍

**Comment by arichornlover** [comment]
Sorry about not using the latest YouTube ipa in the current release. The site where I download ipas is currently down at the moment, so I will be stuck on using the YouTube v18.11.2 ipa until I update y’all about it being up in the replies below.

**Reply by arichornlover** [reply]
Their website is back! From what I believe I had that error on the site for 4 days but glad I can do updates like normal again!

**Comment by dayanch96** [comment]
Hey, thanks. Sure, but there no reason - downloading cant work technically
because Majd removed it from his server.

Чт, 27 апр. 2023 г. в 05:33, arichorn ***@***.***>:

**Reply by dayanch96** [reply]
Looks like dylib name edited after i got email. I’ll do that

**Reply by arichornlover** [reply]
> Hey, thanks. Sure, but there no reason - downloading cant work technically because Majd removed it from his server. Чт, 27 апр. 2023 г. 

I know downloading isn't possible on cercube so I'm adding Reborn as an Alternative way of downloading videos & audio and then modifying the cercube deb with some small little changes and fixes if I'm able to.

**Reply by arichornlover** [reply]
and @dayanch96 heres this piece of code. I expanded the red progress bar option with this.
buffered/unbuffered I believe this is what they are
![IMG_1104](https://github.com/arichorn/CercubePlusExtra/assets/78001398/bdebd7d2-03b3-4d2b-967c-4d70222ebe1b)
and the reason why I done this is because it disappears on the latest version of yt if you don’t interact with the bar. Just like how the Red Progress Bar turns to White. anyways sorry for my awful explaination and the code below hasn’t been tested out yet so I’m sorry for that aswell.

```
// Bring back the old buffered progress bar - @dayanch96
%hook YTSegmentableInlinePlayerBarView
- (UIColor *)_unbufferedProgressBarColor {
return IsEnabled(@"oldBufferedProgressBar_enabled") ? [UIColor colorWithRed: 0.65 green: 0.65 blue: 0.65 alpha: 0.60] : %orig;
}
%end

```

**Reply by dayanch96** [reply]
> and @dayanch96 heres this piece of code. I expanded the red progress bar option with this.
> buffered/unbuffered I believe this is what they are

I've lost the crux of the matter

**Reply by dayanch96** [reply]
oh, guess i got you. if you trying to bring back old BUFFERED progress bar color for dark mode, use this code intead:

```
%hook YTSegmentableInlinePlayerBarView
- (void)setBufferedProgressBarColor:(id)arg1 {
    arg1 = IsEnabled(@"oldBufferedProgressBar_enabled") ? [UIColor colorWithRed:0.65 green:0.65 blue:0.65 alpha:0.60] : %orig;
    %orig;
}
%end
```

**Comment by dayanch96** [comment]
and, for cercube. it started crashing again and i dont see a single reason to use it at all, so you can implement reborn instead, plus its open source

**Reply by arichornlover** [reply]
Alright. Guess I have to remove Cercube.

**Reply by arichornlover** [reply]
Hello! I’m back! I’ve did so much rebranding for the past 5 days! Which Took a ton of effort to get this and was able to get things working. Not only that I modified the Reborn Source Code and I’ll put the link here in case.
https://github.com/arichorn/YouTube-Reborn-Sideloaded
so far things are going great so far.
